src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -174,14 +196,23 @@ public FormValidation doCheckReturnStatus(@QueryParameter boolean returnStdout,
             if (returnStdout) {
                 durableTask.captureOutput();
             }
+            TaskListener listener = context.get(TaskListener.class);
             if (step.encoding != null) {
                 durableTask.charset(Charset.forName(step.encoding));
             } else {
                 durableTask.defaultCharset();
             }
-            controller = durableTask.launch(context.get(EnvVars.class), ws, context.get(Launcher.class), context.get(TaskListener.class));
+            controller = durableTask.launch(context.get(EnvVars.class), ws, context.get(Launcher.class), listener);
             this.remote = ws.getRemote();
-            setupTimer();
+            if (USE_WATCHING) {
+                try {
+                    controller.watch(ws, new HandlerImpl(this, ws, listener), listener);
+                    watching = true;
+                } catch (UnsupportedOperationException x) {
+                    LOGGER.log(Level.WARNING, null, x);
```

Comment:
Shouldn't we fail out here?

Reply:
No, we just fall back to polling mode.

Reply:
Worth a code comment to mention that perhaps?

Reply:
ack